### PR TITLE
Fix(HK): latest file extraction from HK

### PR DESCRIPTION
**Pull request by ivadym**
## Description

Related to https://github.com/Clinical-Genomics/housekeeper/issues/90

### Changed
- Delivery report uploaded to Scout files section

### Fixed
- Extraction of the latest file from HK for delivery reports and Scout uploads

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### Tests

- Current ouput `cg generate delivery-report <case_id>`:
Screenshot 2023-03-29 at 11 46 46

- Now it includes the latest file:
Screenshot 2023-03-31 at 14 48 29

- Upload to scout config file: `cg upload create-scout-load-config <case_id>`
Screenshot 2023-03-29 at 17 03 48

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/report/templates/balsamic_report.html [hunk 1]
```diff
@@ -230,16 +230,16 @@ <h4 class="card-title">Scout</h4>
...
+                  <li><strong>Förvärvade SNVs och INDELs</strong>: <em>{{ case.data_analysis.scout_files.snv_research_vcf.replace(case.id, case.name) }}</em></li>
                 {% endif %}
                 {% if case.data_analysis.scout_files.sv_vcf != 'N/A' %}
-                  <li><strong>Clinical somatic structural variants fil</strong>: <em>{{ case.data_analysis.scout_files.sv_vcf.replace(case.id, case.name) }}</em></li>
+                  <li><strong>Kliniskt relevanta förvärvade SVs</strong>: <em>{{ case.data_analysis.scout_files.sv_vcf.replace(case.id, case.name) }}</em></li>
                 {% endif %}
                 {% if case.data_analysis.scout_files.sv_research_vcf != 'N/A' %}
-                  <li><strong>Research somatic structural variants fil</strong>: <em>{{ case.data_analysis.scout_files.sv_research_vcf.replace(case.id, case.name) }}</em></li>
+                  <li><strong>Förvärvade SVs</strong>: <em>{{ case.data_analysis.scout_files.sv_research_vcf.replace(case.id, case.name) }}</em></li>
```
Comment (ivadym) [inline cg/meta/report/templates/balsamic_report.html:242]: Could you have a look at these translations in the delivery report @pbiology? Basically it shows which files from Caesar have been uploaded to Scout 

Screenshot 2023-03-29 at 13 24 34

File: cg/apps/housekeeper/hk.py [hunk 2]
```diff
@@ -158,6 +158,13 @@ def get_files(
         """
         return self._store.get_files(bundle_name=bundle, tag_names=tags, version_id=version)
 
+    def get_latest_file(
+        self, bundle: str, tags: Optional[list] = None, version: Optional[int] = None
+    ) -> Optional[File]:
+        """Return latest file from Housekeeper, filtered by bundle and/or tags and/or version."""
+        files: Query = self._store.get_files(bundle_name=bundle, tag_names=tags, version_id=version)
```
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/apps/housekeeper/hk.py:164]: I agree ⭐

**Comment by ivadym** [comment]
> Nice 👍 I guess we should try to keep all logic separated to the housekeeper repo, so maybe we should add a get_latest_file function to that repo later

Thanks for the review @seallard 🏅 , I have also updated the file extraction for Scout uploads to take the latest file FYI: [38c211e](https://github.com/Clinical-Genomics/cg/pull/1868/commits/38c211ed2d9905d901c185e79ac5f9628de36a73)

File: cg/meta/report/templates/balsamic_report.html (hunk 1 above)
Comment (pbiology) [reply cg/meta/report/templates/balsamic_report.html:242]: Should we call the research one "all SNVs and Indels"? To further explain how it differs from the clinical ones?

**Review by pbiology — COMMENTED** [review]
Swedish looks good. 

Just the one question about how to distinguish the research and clinical results

**Comment by ivadym** [comment]
> Swedish looks good. 
> 
> Just the one question about how to distinguish the research and clinical results
> Should we call the research one "all SNVs and Indels"? To further explain how it differs from the clinical ones?

Maybe we need something to make the distinction, but I'm not sure what is the best option. The research files does not contain all the variants, they are being filtered against SWEGEN and GNOMAD... what do you think?

**Comment by pbiology** [comment]
> Förvärvade SVs

What about "Förvärvade SVs för forskning"?

**Comment by ivadym** [comment]
> > Förvärvade SVs
> 
> What about "Förvärvade SVs för forskning"?


I like it, thanks for having a look at this @pbiology  🏅

**Pull request merged**